### PR TITLE
Fix login redirect issues

### PR DESCRIPTION
**Pull request by jimchamp**
Closes #

This contains a few login-related bug fixes:
1. By default, patrons will be redirected to their loans page if no other redirect URL is POSTed.
2. Replaced incorrect `redir` query parameter used by the book page reading log dropper.
3. Do not render book note link if patron is not logged in (the link doesn't function for logged-out patrons).

### Technical

We failed to account for the hidden `redirect` input in the login template in #5774.  This was populated by default with the `HTTP_REFERER` by the `/login` GET method.  `/account/loans` now replaces `HTTP_REFERER` as the default redirect value.

The main issue with this approach is that using the navbar login link on a book page will result in a redirect to the loans page.  Some may consider this behavior to be unexpected/undesired.  To remedy this, we could chose a sensible default based on the referer's path in the login `GET` method if no redirects are given.

### Testing

While logged out:
1. Navigate to a book page.
2. Ensure that the book note link is not visible below the star rating component.
3. Click on the reading log dropper.  This will load the login page.
4. Log into the site, ensuring that you have been redirected to the book page.
5. Ensure that the book note link is present.
6. Log out and log back in while in different contexts (from Google search, from the homepage, etc.).  Ensure that the redirect behavior is as expected.

### Stakeholders

@mekarpeles

### CODE REVIEW

**Comment by jimchamp** [comment]
This may be too sweeping of a change.  We may still want to use the referer when the navbar login link is used in specific contexts (book pages, author pages, search results pages, etc.).

File: openlibrary/plugins/upstream/account.py [hunk 1]
```diff
@@ -375,7 +372,7 @@ def POST(self):
             connect=None,
             password="",
             remember=False,
-            redirect='/',
```
Comment (mekarpeles) [inline openlibrary/plugins/upstream/account.py:378]: this is getting expanded to the full host on e.g. /login

**Comment by jimchamp** [comment]
@mekarpeles after implementing the changes that we've tested today, I still see that we redirect to `/account/loans` when logging in from a books page with navbar "Log in" link.

**Comment by mekarpeles** [comment]
If we're on a books page and we explicitly click login, then I think we should send them to `/account/loans`

**Pull request merged**